Notes.md

```diff
@@ -0,0 +1,251 @@
+* Draw a flowchart to represent the connections between webpages at: <http://i.imgur.com/kgKyxrH.png>
+* Design the file structure. We mimic the style at: <http://gouthamanbalaraman.com/blog/flask-app-directory-structure.html>
+
+2.Get started
+* We begin to write Python for schema.py and HTML/CSS(using CSS boostrap) code for the design of the main page.  
+
+##Part3 Study Guide
+We work through my demo, which is built on top of Ross's http-demo
+###0.To understand Python,CSS,HTML,Javascript(JQuery) files in general, please check out the references in Part4.
+
+###1.To understand code in: `settings.cfg config.py db.py`
+* Read the first part of the documentation of configuration in python3 at: <https://docs.python.org/3/library/configparser.html>
+* Read the first part of the documentation of sessionmaker at: <http://docs.sqlalchemy.org/en/latest/orm/session_api.html#session-and-sessionmaker>
+
+###2.To understand code in: `main.py schema.py`
+* Read a quick start tutorial for Flask at: <http://flask.pocoo.org/docs/0.10/quickstart/#a-minimal-application>
+* Read a tutorial for Flask-Alchemy at: <http://docs.sqlalchemy.org/en/latest/orm/tutorial.html>
+* Skim through the documentation of enum at: <https://docs.python.org/3/library/enum.html>
+
+###3.To understand code in: `dataforms.py`
+* Read a short tutorial of WTForms at <http://wtforms.readthedocs.io/en/latest/crash_course.html> <br>
+* Read WTForms in Flask at: <http://flask.pocoo.org/docs/0.10/patterns/wtforms/>
+
+###4.To understand code in: `create_lab.html`
+* One specific use of JQuery at <https://www.codecademy.com/courses/web-beginner-en-v6phg/2/3?curriculum_id=50a3fad8c7a770b5fd0007a1>
+
+
+##Part4 References
+###1.Tools
+####1.general development environment
+* A good intro at: <https://github.com/reed-college/2016_sds_lesson_notes/blob/master/lesson_01_development_environment.markdown>
+
+* virtualenv/virtualenvwrapper and some basic terminal commands at https://github.com/reed-college/2016_sds_lesson_notes/blob/master/lesson_03_beginning_development.markdown
+
+####2.github
+* A quick-start at: <https://guides.github.com/activities/hello-world/>
+* A useful cheat-sheet at: <https://education.github.com/git-cheat-sheet-education.pdf>
+* A relatively complete intro to git/github at: <https://github.com/reed-college/2016_sds_lesson_notes/blob/master/lesson_02_git.markdown>
+
+####3.ssh
+* Notes at: <https://github.com/reed-college/2016_sds_lesson_notes/blob/master/lesson_04_ssh_plain_text.markdown>
+
+####4.network
+* Basic concepts of http/html at: <https://github.com/reed-college/2016_sds_lesson_notes/blob/master/lesson_05_http_and_html.markdown> 
+
+####5.database/ORM
+* Basic concepts of database(Postgresql) and ORM at: <br>
+<https://github.com/reed-college/2016_sds_lesson_notes/blob/master/lesson_06_databases_part_one.markdown>
+<https://github.com/reed-college/2016_sds_lesson_notes/blob/master/lesson_06_databases_part_two.markdown>
+
+
+###2.Language related
+####1.Python3
```

Comment:
general, language level resources aren't really appropriate for a project repo, unless the project is didactic. Additionally, the fact that github's syntax highlighting is failing for this file indicates that there's probably a markdown syntax error. Keeping the markdown valid isn't strictly necessary, since the file will often render just fine anyway, but it is kinda of a niceness for future writers, who might face weird bugs in rendering because of an existing error.